ci/build-and-test-nupic-bindings.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have purchased from
+# Numenta, Inc. a separate commercial license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+set -o errexit
+
+
+USAGE="Usage:
+
+[BUILD_TYPE=Release | Debug] [RESULT_KEY=key-string] [WHEEL_PLAT=platform] $( basename ${0} )
+
+This script builds and tests the nupic.bindings Python extension.
+
+In Debug builds, also
+  - Turns on the Include What You Use check (assumes iwyu is installed)
+
+ASUMPTION: Expects a pristine nupic.core source tree without any remnant build
+   artifacts from prior build attempts. Otherwise, behavior is undefined.
+
+
+INPUT ENVIRONMENT VARIABLES:
+
+  BUILD_TYPE : Specifies build type, which may be either Release or Debug;
+               defaults to Release. [OPTIONAL]
+  RESULT_KEY : Build result key; used to decorate artifact names. [OPTIONAL]
+  WHEEL_PLAT : Wheel platform name; pass manylinux1_x86_64 for manylinux build;
+               leave undefined for all other builds.
+
+OUTPUTS:
+  nupic.bindings wheel: On success, the resulting wheel will be located in the
+                        subdirectory nupic_bindings_wheelhouse of the source
+                        tree's root directory.
+
+  test results: nupic.bindings test results will be located in the subdirectory
+                test_results of the source tree's root directory with the
+                the following content:
+
+                cplusplus-test-results.txt
+                junit-test-results.xml
+                htmlcov-report/
+
+"
+
+if [[ $1 == --help ]]; then
+  echo "${USAGE}"
+  exit 0
+fi
+
+if [[ $# > 0 ]]; then
+  echo "ERROR Unexpected arguments: ${@}" >&2
+  echo "${USAGE}" >&2
+  exit 1
+fi
+
+
+set -o xtrace
+
+
+# Apply defaults
+BUILD_TYPE=${BUILD_TYPE-"Release"}
```

Comment:
@scottpurdy:
> Can you explain this line?

You mean my comment "Apply defaults" didn't do the trick? 😸 

This is standard bash syntax for a default fallback value. Since BUILD_TYPE is documented as optional, defaulting to Release, we need to apply the default value if one wasn't passed.